spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+            + $"/databases/{databaseId}";
+            // Gets a transaction object that captures the database state
+            // at a specific point in time.
+            using (TransactionScope scope = new TransactionScope(
+                TransactionScopeAsyncFlowOption.Enabled))
+            {
+                // Create connection to spanner.
+                using (var connection = new SpannerConnection(
+                    connectionString))
+                {
+                    var cmd = connection.CreateSelectCommand(
+                        "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                    // Read #1.
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Console.WriteLine(
+                                $"SingerId : {reader["SingerId"]}"
+                                + $" AlbumId : {reader["AlbumId"]}"
+                                + $" AlbumTitle : {reader["AlbumTitle"]}");
+                        }
+                    }
+                    // Read #2. Even if changes occur in-between the reads, 
+                    // the transaction ensures that Read #1 and Read #2 
+                    // return the same data.
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Console.WriteLine(
+                                $"SingerId : {reader["SingerId"]}"
+                                + $" AlbumId : {reader["AlbumId"]}"
+                                + $" AlbumTitle : {reader["AlbumTitle"]}");
+                        }
+                    }
+                }
+                // Commit the transaction to Spanner table.
+                scope.Complete();
+                // Yield Task thread back to the current context.
+                await Task.Yield();
+                return;
+                // [END read_only_transaction]
+            }
+        }
+
+        public static async Task ReadSampleDataAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data]
+            // [START read_data]
+            Console.WriteLine("Querying data...");
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                Console.WriteLine("Connected...");
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())

```

Comment:
this is the same as the query one right?
maybe we can change it up a little bit and show how you can do parallel queries by running executereaderasync from several commands on the same connection... ?
you can do multiple reads and writes concurrently on the same connection -- but concurrent writes will consume extra sessions.